sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/keyvault/PostLegacyPropertyEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.keyvault;
+
+import com.azure.spring.autoconfigure.unity.AbstractLegacyPropertyEnvironmentPostProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.util.CollectionUtils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import static org.springframework.core.env.StandardEnvironment.SYSTEM_ENVIRONMENT_PROPERTY_SOURCE_NAME;
+
+/**
+ * Convert legacy properties in Key Vault property sources to the current and set into environment after
+ * {@link KeyVaultEnvironmentPostProcessor}.
+ *
+ * Due to that properties in Key Vault has higher precedence than those in local, thus when comparing whether legacy
+ * properties in Key Vault should be converted, they only need to be compared those in Key Vault.
+ *
+ * The converted current properties should have higher precedence than those in local, so the precedence is set
+ * to be lower only than system properties, which is refered to the behavior of Key Vault property sources.
+ */
+public class PostLegacyPropertyEnvironmentPostProcessor extends AbstractLegacyPropertyEnvironmentPostProcessor {
+
+    public static final int DEFAULT_ORDER = KeyVaultEnvironmentPostProcessor.DEFAULT_ORDER + 1;
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostLegacyPropertyEnvironmentPostProcessor.class);
+
+    @Override
+    public int getOrder() {
+        return DEFAULT_ORDER;
+    }
+
+    /**
+     * When only legacy properties are detected from each key vault property source, convert legacy properties to
+     * the current, and create a new {@link Properties} to store the converted current properties of each key vault
+     * property source.
+     *
+     * @param legacyToCurrentMap A {@link Properties} contains a map of all legacy properties and associated current properties.
+     * @return A {@link Properties} to store mapped current properties
+     */
+    @Override
+    protected Properties convertLegacyPropertyToCurrent(Properties legacyToCurrentMap) {
+        Properties properties = new Properties();
+        List<KeyVaultPropertySource> propertySourceList = getKeyVaultPropertySourceList();
```

Comment:
how about this:

1. getKeyVaultPropertySourceList returns a Stream ( change the method name as well )
2. Stream.sorted(Collections.reverseOrder())

personally i think this has more readability